numpy/core/src/umath/ufunc_type_resolution.c

```diff
@@ -288,7 +288,7 @@ PyUFunc_DefaultTypeResolver(PyUFuncObject *ufunc,
     } else {
         /* Find the specified ufunc inner loop, and fill in the dtypes */
         retval = type_tuple_type_resolver(ufunc, type_tup,
-                        operands, casting, any_object, out_dtypes);
+                        operands, input_casting, casting, any_object, out_dtypes);
```

Comment:
I'm confused - I see `input_casting` added to two functions, but not used!?

Reply:
Its passed on to `ufunc_loop_matches` where before `casting, casting` was passed on its, not `input_casting, casting`.

Reply:
OK, makes sense, thanks!